client/environment.go

```diff
@@ -46,27 +46,8 @@ func (c *APIClient) ReadEnvironment(environmentID string) (*models.Environment,
 
 func (c *APIClient) UpdateEnvironment(req models.UpdateEnvironmentRequest) (string, error) {
 	var resp models.CreateJobResponse
-	if err := c.client.Put("/environment", req, &resp); err != nil {
-		return "", err
-	}
-
-	return resp.JobID, nil
-}
-
-func (c *APIClient) CreateLink(req models.CreateEnvironmentLinkRequest) (string, error) {
-	var resp models.CreateJobResponse
-	path := fmt.Sprintf("/environment/%s/link", req.SourceEnvironmentID)
-	if err := c.client.Post(path, req, &resp); err != nil {
-		return "", err
-	}
-
-	return resp.JobID, nil
-}
-
-func (c *APIClient) DeleteLink(req models.DeleteEnvironmentLinkRequest) (string, error) {
-	var resp models.CreateJobResponse
-	path := fmt.Sprintf("/environment/%s/link/%s", req.SourceEnvironmentID, req.DestEnvironmentID)
-	if err := c.client.Delete(path, nil, &resp); err != nil {
+	path := fmt.Sprintf("/environment/%s", req.EnvironmentID)
```

Comment:
`EnvironmentID` needs to be removed from this object. The `UpdateEnvironment` signature should be: 
```
UpdateEnvironment(environmentID string, req models.UpdateEnvironmentRequest) (string, error)
```

Reply:
Not sure what you mean. It doesn't make sense to remove the entity id in each of the entity update request models? It is consistent with all the other Update functions in the current API refactor branch:
```
type Client interface {
	UpdateEnvironment(req models.UpdateEnvironmentRequest) (string, error)
	UpdateLoadBalancer(req models.UpdateLoadBalancerRequest) (string, error)
	UpdateService(req models.UpdateServiceRequest) (string, error)
        ...
}
```

I think we might've visited this with the original discussion with env links change. If we remove the entity id, we would need to create job request specific models. Which would be super tedious to translate from one model to another almost identical looking model just for the job runner.

Reply:
right - we hit on this in slack quite a bit so will refer to that

Reply:
Yes, we could do this to be more RESful but PATCH requests would need to be constructed differently to the PUT requests as a PATCH contains a description of the changes rather than the changed state i.e., the request for adding a link would contain just the addition and look something like
```
[
  {
    "action": "addLink",
    "link": "environment_id"
  }
]

```

That would require significant changes server-side and as you mentioned previously, add more responsibilities to the Terraform Provider. Instead we could change the path to the below to keep supporting PUT and be more RESTful.
Instead of 
```
PATCH /environment/:id
{}
```

```
PUT /environment/:id/links/
["eid1", "eid2", "eid3"]

PUT /environment/:id/minclustercount/
{"1"}
```

And could lead to:
```
type UpdateEnvironmentLinksJob struct {
    EnvironmentID string
    Req                   []string
}

type UpdateEnvironmentMinCountJob struct {
    EnvironmentID string
    Req                   int
}
```

Making it more complicated by trying to be more RESTful? @zpatrick

Reply:
hmmm - we need to also consider keeping this consistent w/ the other entity updates. What if we we keep the method + paths you laid out, but the controller layer does the conversion of turning those request into generic`UpdateEntityJob` structs (which are just renamed versions of `UpdateEntityRequest`). That way we are keeping the `EntityUpdate` method the same in the `provider` layer.

Reply:
Sure - I think that isn't too much responsibility at for the Controller to handle. ✅